campaignion_wizard/src/WebformStep.php

```diff
@@ -33,8 +33,42 @@ public function pageCallback() {
 
     // Build form for webform_template select box.
     if (module_exists('campaignion_action_template')) {
+      // Check if there are existing form submissions.
+      $submissions_exist = FALSE;
+      $sql = <<<SQL
+SELECT count(*) AS submissions
+FROM {webform_submissions}
+WHERE nid=:nid
+SQL;
+      $submissions = db_query($sql, array(':nid' => $this->wizard->node->nid));
+      foreach ($submissions as $s) {
+        if ($s->submissions > 0) {
+          $submissions_exist = TRUE;
+        }
+      }
+
       $build[] = drupal_get_form('campaignion_action_template_selector_form', $this->wizard->node)
-        + array('#weight' => -99);
+        + array('#weight' => 1);
+
+      if ($submissions_exist) {
+        $build[1]['webform_template']['webform_template_warning'] = array(
+          '#markup' => '<div class="action-template-warning">' .
+          t('Applying form templates is not possible on forms that have already captured data. If you want to apply a form template you have to <a href="@delete-submissions">delete all form data first</a>.',
+            array(
+              '@delete-submissions' => url(
+                'node/' . $this->wizard->node->nid . '/webform-results/clear',
+                array(
+                  'query' => array(
+                    'destination' => 'node/' . $this->wizard->node->nid . '/wizard/form'
+                  ),
```

Comment:
Please use the new style array syntax for new code.

Also splitting this into multiple statements would avoid such deeply nested arrays. Also you can perhaps use `$nid` here as you do further up.

```php
$o['query']['destination'] = url("node/$nid/wizard/form");
$url = url("node/$nid/webform-results/clear", $o);
$warning = t("…", ['@delete-submissions' => $url]);
```